rfc/2022/2022-08-23_CIT_hostnames.md

```diff
@@ -0,0 +1,246 @@
+
+Reference: [Complete list of Datadog Tests](https://vagov.ddog-gov.com/synthetics/tests?query=tag%3A%28%22team%3Acit%22%29)
+
+
+#### **Vets-website: Change \*api.va.gov References in Unit and Integration Tests**
+We have updated `api.va.gov` references to `platform-api.va.gov` in the test files of the vets-website repository. These updates to the test references did not break any unit or integration tests, as they were in many cases part of canned API responses and not used in any test assertions.
+
+Certain test references to `sandbox-api.va.gov` and `api.va.gov` did not require any changes because these reference endpoints will still be served by the Lighthouse API. **Example**: `health-care-questionnaire` application (not used in Production.)
+
+Parameterizing the API URL was done where possible. Setting the API_URL constant as an environment variable will ensure that unit specs that check for the variable value will pass before and after changing the configured API URL to the new *platform-ap.va.gov hostname.
+
+To prevent the future use of *api.va.gov in test files, the Release Tools Team should add custom linting rules that instruct users to use the new API URLs.
+&nbsp;
+&nbsp;
+#### **Diagram: New Production State with vets-api in BRD and Lighthouse/Kong**
+
+![Prod-release-state-BRD-Kong](https://user-images.githubusercontent.com/1426601/187307838-1ba75418-49e9-479c-8c85-89e8117751eb.png)
+&nbsp;
+&nbsp;
+#### **Diagram: Future Production State with vets-api in EKS and Lighthouse/Apigee**
+
+![Prod-release-state-EKS-Apigee](https://user-images.githubusercontent.com/1426601/187308803-f87cd357-d3ba-497a-94c6-ea792769ee6a.png)
+&nbsp;
+&nbsp;
+### Pre-Release Tasks
+
+#### **Coordination**
+  - Kickoff Meeting to discuss team roles and responsibilities:
+    - Release Plan timeline.
+    - Rollback plan.
+    - Notification strategy:
+      - Team notifications.
+      - User notifications for potential login availability.
+    - Downtime window.
+      - In order to minimize disruptions to Staging and Production, a coordinated release of both applications should consider the different build and deploy methods and runtimes.
+      - Based on historic runtimes, determine the optimal timing for triggering the deployment of both vets-api and vets-website so that they finish at approximately the same time.
+
+
+#### **VA**
+   * Submit required notifications to the VA so that entities, .e.g., the Help Desk, are aware of any planned downtime or interruption of services.
+
+
+#### **Site Reliability Engineering**
+   * SRE playbooks prepared to use the *platform-api.va.gov hostname.
+
+
+#### **Release Tools Team**
+  - Adds linting rules to prevent use of the api.va.gov hostname.
+
+
+### Release Steps
+Due to the interdependence of API calls made from VA.gov and authentication cookies set by vets-api, the deployment of the new *platform-api.va.gov hostname will require the coordinated releases of vets-api and vets-website. 
```

Comment:
Is it possible to have authentication cookies that include multiple hostnames? Can vets-api be configured to configure cookie based on requesting domain?

Reply:
Cookies can't be assigned to multiple hostnames. Setting the cookie based on the requesting domain won't help with the need to coordinate what the frontend is configure to use with the auth callbacks issued by IAM/ID.me. During whatever interval those two don't match, sign-in will be broken. 

You *can* have a domain-scoped cookie - i.e. one for `.va.gov` - in which case the browser would send the cookie for subsequent requests to _any_ va.gov subdomain. That is in fact what was done a couple years ago when there was a one-off SSO implementation exclusively between MHV and VA.gov. 

What I don't know is if there was any additional VA-wide (e.g. NSOC) approval needed to have that in place.

Reply:
You could also *potentially* try setting up an extra internal redirect between the auth callback and the final redirect to www.va.gov that would hop the domain back to whatever the frontend was expecting, and decouple these two things. This would require some prototyping in dev to see if it worked. 

(We checked and IAM is only capable of having one SAML callback endpoint configured at a time, unlike some other auth providers).

Reply:
I added some additional details about why the releases need to be coordinated, and also all the alternatives that were considered. @patrickvinograd let me know if I misrepresented anything and I can edit.